### PR TITLE
Frontend tests

### PR DESCRIPTION
**Pull request by hirunatan**
> Please provide enough information so that others can review your pull request:

> Explain the **details** for making this change. What existing problem does the pull request solve?

> Screenshots/GIFs

### CODE REVIEW

File: frontend/src/app/util/globals.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+ * Provide instances of some javascript global objects, by looking for
+ * them in the browser, the current worker or creating a dummy for test
+ * environment.
+ */
+
+"use strict";
+
+goog.provide("app.util.globals");
+goog.provide("app.util.globals.window");
+goog.provide("app.util.globals.location");
```
Comment (niwinz) [inline frontend/src/app/util/globals.js]: this line and the previous one, is not necessary (window and location), only the first one is mandatory

File: frontend/src/app/util/globals.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+goog.scope(function() {
+  var win;
+  if (typeof window !== "undefined") {
```
Comment (niwinz) [inline frontend/src/app/util/globals.js]: don't acces window **as-is**; use `goog.global.window`

File: frontend/src/app/util/globals.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  } else if (typeof self !== "undefined"){
+    win = self;
+  } else {
+    win = {};
+  }
+
+  app.util.globals.window = win;
+
+  var loc;
+  if (typeof location !== "undefined") {
```
Comment (niwinz) [inline frontend/src/app/util/globals.js]: the same as for `window`

File: frontend/src/app/util/globals.js (hunk 1 above)
Comment (hirunatan) [reply frontend/src/app/util/globals.js]: done

File: frontend/src/app/util/globals.js (hunk 3 above)
Comment (hirunatan) [reply frontend/src/app/util/globals.js]: done

File: frontend/src/app/util/globals.js (hunk 2 above)
Comment (hirunatan) [reply frontend/src/app/util/globals.js]: done

**Pull request merged**